src/plugins/data_source_management/public/components/utils.ts

```diff
@@ -54,6 +55,44 @@ export async function getDataSourcesWithFields(
   return response?.savedObjects;
 }
 
+export function getDefaultDataSource(
```

Comment:
Would be nice to add some test cases for this function :)

Reply:
Maybe leave a comment to describe the default data source selection logic?

Reply:
Sure. It is a draft CR. Would add a comment and test case